cluster-autoscaler/core/scaledown/actuation/actuator.go

```diff
@@ -69,9 +69,7 @@ func NewActuator(ctx *context.AutoscalingContext, csr *clusterstate.ClusterState
 // CheckStatus should return an immutable snapshot of ongoing deletions. Before the TODO is addressed, a live object
 // is returned instead of an immutable snapshot.
 func (a *Actuator) CheckStatus() scaledown.ActuationStatus {
-	// TODO: snapshot information from the tracker instead of keeping live
-	// updated object.
-	return a.nodeDeletionTracker
+	return a.nodeDeletionTracker.Snapshot()
```

Comment:
I think ideally we'd just replace `ActuationStatus` interface with a simple struct that doesn't have the mutex overhead. It would also be simpler to mock in tests. Can you add a TODO for this somewhere? Perhaps next to the interface definition.